### DISCUSSION TITLE
soup fails (sort of), elastalert sometimes missing. Status code 502

### DISCUSSION THREAD

**Discussion by Sazerak** [Security-Onion-Solutions/securityonion#8312] [Unsupported Versions] [answered]
Since roughly .100, I've been getting gremlins when updating SO2. I'll start by mentioning that around that time I changed the hardware I was running on from an old DL380 G7 to a (less) old desktop setup running an i5-6600k.

I noticed a while back that sometimes updating would fail (I don't really recall the reason, newborn dad brain, but I seem to remember it would hit 300 attempts to reach/connect to something, maybe kibana). Eventually after a couple more soup commands, all would be good.

Edit: Found the section in the logs:
```
Waiting for value 'Elastic' at 'http://localhost:5601/app/kibana' (300/300)
Server is not ready
Server still not ready after 300 attempts; giving up.
```

```
Waiting for value 'green open' at 'https://localhost:9200/_cat/indices/.kibana*' (300/300)
Server is not ready
Server still not ready after 300 attempts; giving up.
```

Fast forward to .120 and elasticsearch would fail after a number of days. I did a discussion (https://github.com/Security-Onion-Solutions/securityonion/discussions/7810) and it seems it was likely related to lack of RAM. Increased the RAM, and everything seemed fine after that (I'm really only mentioning this one in case it could be related to my new issue below).

Yesterday, I updated to .130 and once again soup failed. Same kibana 300 attempt error. However, when I would run soup again to try and "fix" the install, it would just tell me I'm up to date and then quit.

Now when I boot SO, it seems to mostly boot fine (at least as far as kibana is concerned), but about 50% of the time the elastalert container is missing (I checked with docker ps and confirmed the container wasn't created but the image exists). Sometimes I'll reboot and then the container is there. But when the container is there and I try to check anything on the webpage (alerts, dashboard, hunt), I get the following message:

```
Request failed with status code 502
See the Help section of the Security Onion documentation for additional troubleshooting guidance.
```

I've never seen this message and couldn't find any reference to status code 502 in the docs (although maybe that's my newborn-dad-lack-of-sleep-brain playing tricks on me)

Any guidance on troubleshooting this would be appreciated.

**Comment by dougburks** [comment]
> Now when I boot SO, it seems to mostly boot fine (at least as far as kibana is concerned), but about 50% of the time the elastalert container is missing (I checked with docker ps and confirmed the container wasn't created but the image exists). Sometimes I'll reboot and then the container is there.

It may be possible that this is just a timing issue and the service eventually comes up. How long have you waited before checking?

Have you modified your Elastalert or Playbook configuration in any way? If so, have you tried reverting that back to defaults?

> Request failed with status code 502

Check the Elasticsearch logs as shown at https://docs.securityonion.net/en/2.3/help.html.

**Reply by Sazerak** [reply]
Hi Doug, thanks for the reply. I've waited up to ~10 minutes and the missing container didn't start up. Does the system periodically check for missing services and attempt to restart them?

I haven't modified anything in Elastalert or Playbook. My system is pretty well stock (standalone install).

I just looked through the elasticsearch logs and found many warnings. They're all pretty well identical and start with:
```
[2022-07-18T14:21:47,270][WARN ][rest.suppressed          ] path: /elastalert_status/_search, params: {size=1000, index=elastalert_status}
org.elasticsearch.action.search.SearchPhaseExecutionException: all shards failed
     at (bunch of stuff here)
```
and end with:
```
Caused by: org.elasticsearch.action.NoShardAvailableActionException
     at (bunch of stuff here)
```

Could this be related to memory again? I increased my memory from 16GB to 20GB after my last discussion and it seemed to have fixed the issue until the update to .130. Does .130 require more memory?

**Reply by isaacgolding** [reply]
20GB of ram could very well not be enough memory.   I have 80GB allocated to my SO install and I average 60% of its memory in use.    When I ran under 32GB or less of memory I sometimes had problems similar to what you are describing.  My swap usage under 32GB of ram was high which slowed things down dramatically and cause problems.     I'm monitoring 50-70 IP addresses with an average network throughput of around 10Mbs,   

When initially testing SO in my VM system I started on the min specs as listed on the docs and kept increasing CPU cores and Memory until I saw the system running smoothly.  It took me a few weeks to get to where I am at now without weird problems cropping up and the system not being responsive.   

If you are running in a VM I would suggest giving SO every bit of free ram you have available to it.   You will find that when you get your used memory % (as listed in Grafna) down to about 60% the system will use the swap partition less and your system will likely calm down.

How many "devices" are you monitoring?   What does Grafna show for the following:   average Monitor inbound?  I/O wait?  % memory in use? 

What is the output of sudo so-status?

**Comment by Sazerak** [comment, marked as answer]
So I ended up doing a fresh install since nothing seemed to work and the issue had persisted through multiple updates, so I figured it would be hard to track down. I did up my RAM to 32GB (max I could spare) and changed my /nsm from a HDD to a SSD since I noticed a note in the docs that reads "We recommend placing all Elastic storage (/nsm/elasticsearch) on SSD or fast spinning disk in a RAID 10 configuration." Maybe it'll help, not sure.

For now I consider this issue resolved, but I can't really say it's "solved". Rather it's been mitigated (at least temporarily, time will tell).
